tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1691 @@
+    if ($resultSize < 0) {
+      print_r($this->searchPage->getAndClearMessages());
+      throw new Exception("Couldn't find a search result size on page.");
+    }
+
+    // Now compare to the expected number.
+    if ($this->searchPage->getExpectedSearchResultSize() != 0) {
+      if ($this->searchPage->getExpectedSearchResultSize() != $resultSize) {
+        throw new Exception("Did not find the expected number of posts. (Found: " . $resultSize . ". Expected:" . $this->searchPage->getExpectedSearchResultSize() . ")");
+      }
+    }
+    else {
+      throw new Exception("An expected number was never set. Use facets to set it.");
+    }
+  }
+
+  /**
+   * Implement step to check the pagination is correct on all pages.
+   *
+   * This is meant only to be used after a multipage search when paging is in place.
+   * It goes through the latest stored search result and finds a random page which is accessible on the
+   * currently displayed page, and go to that one.
+   * Notice that the way to move to a particular page is to go to the link /search/thing/<searchcrit>?page=n
+   * where n = 1 yields page 2 in the search result (without ?page it defaults to first page).
+   * Also notice that not all pages are given as options for the user.
+   * On a 10-page result, only 1 + 2 is displayed from the beginning.
+   * From page 3 you get "første" + "forrige". On page 1 and 2 you don't.
+   *
+   * @Then I check pagination on all pages
+   *
+   * @throws Exception
+   *   If errors happens.
+   */
+  public function checkPaginationOnAllPages() {
+    try {
+      $this->check($this->searchPage->checkPaginationOnAllPages(), $this->searchPage->getAndClearMessages());
+    }
+    catch (Exception $e) {
+      throw new Exception("Failed: " . $e->getMessage());
+    }
+    catch (\Page\Exception $e) {
+      throw new Exception("Failed getting pages: " . $e->getMessage());
+    }
+  }
+
+  /**
+   * Type text character by character, with support for newline, tab as \n and \t
+   *
+   * @param string $text
+   *    The text to enter into the field.
+   * @param string $field
+   *    The popular name for the field to enter data into.
+   *
+   * @When I enter :text in field :field
+   *
+   * @throws Exception
+   *    In case of errors.
+   */
+  public function enterTextIntoField($text, $field) {
+    $found = $this->getPage()->find('css', $this->translateFieldName($field));
+    if (!$found) {
+      throw new Exception("Couldn't find the field " . $field);
+    }
+    $this->scrollTo($found);
+    // Click so we place the cursor in the field.
+    $found->click();
+
+    /*
+     * Now it becomes technical, because we will type each character in the $text variable one at a
+     * time, but also we want to use the escape option of f.ex. \n. So we remember if we get the \ char
+     * and then check the next character.
+     */
+    $escaped = false;
+    for ($i = 0; $i < strlen($text); $i++) {
```

Comment:
strlen($text) should be called outside the loop declaration

Reply:
Fixed